src/main/java/seedu/duke/Storage.java

```diff
@@ -0,0 +1,179 @@
+     *            corresponds with the index of the task in the task list to be deleted.
+     * @throws IOException if an error occurs while writing to the file.
+     */
+    public static void deleteFromFile(int num) throws IOException {
+        Scanner s = new Scanner(f);
+        String newData = "";
+        int count = 0;
+        while (s.hasNextLine()) {
+            String temp = s.nextLine();
+            if (count != num) {
+                newData += temp + "\n";
+            }
+            count++;
+        }
+        FileWriter fw = new FileWriter(f);
+        fw.write(newData);
+        fw.close();
+    }
+
+    /**
+     * Updates the marking information stored in the file by rewriting a copy of the entire file to the
+     * same file path and writing the new line of information where the marking has changed.
+     *
+     * @param num the line of information in the file that is to be updated, corresponds with
+     *            the index of the task in the task list whose marking is to be updated.
+     * @param textToReplace the new text to replace the old line of code, which contains
+     *                      the newly updated marking information. Taken from the Task objects toStore() function.
+     * @throws IOException if an error occurs while writing to the file.
+     */
+    public static void changeMarking(int num, String textToReplace) throws IOException {
+        Scanner s = new Scanner(f);
+        String newData = "";
+        int count = 0;
+        while (s.hasNextLine()) {
+            String temp = s.nextLine();
+            if (count != num) {
+                newData += temp + "\n";
+            } else {
+                newData += textToReplace;
+            }
+            count++;
+        }
+        FileWriter fw = new FileWriter(f);
+        fw.write(newData);
+        fw.close();
+    }
+
+    public static void findFromFile(String input) throws IOException{
+        Scanner s = new Scanner(f);
+        String lowercaseInput = input.toLowerCase();
+        int ptr = 1;
+        int count = 0;
+        while (s.hasNextLine()) {
+            if (s.nextLine().toLowerCase().contains(lowercaseInput)) {
+                count++;
+                if (count == 1) {
+                    UI.printTop("      Here are the matching tasks in your list:\n");
+                }
+                taskList.getTask(ptr - 1).printMatchDesc(count);
+            }
+            ptr++;
+        }
+        //if no matching tasks found
+        if (count == 0) {
+            UI.printOut("      There was no matching tasks found in your list. Perhaps a typo?\n");
+        } else {
+            UI.printLine();
+        }
+    }
+
+    /**
+     * Writes the information of the newly added task to the file
+     *
+     * @param task the Task object whose information is to be stored.
+     */
+    public static void add(Task task) { //to append items to taskList
```

Comment:
Would it be better if the method name is more detailed to reflect the function of adding task to file?
One possible suggestion is "addTaskToFile"